dashboard/test/models/script_test.rb

```diff
@@ -7,19 +7,31 @@ class ScriptTest < ActiveSupport::TestCase
   self.use_transactional_test_case = true
 
   setup_all do
+    Rails.application.config.stubs(:levelbuilder_mode).returns false
     @game = create(:game)
     @script_file = File.join(self.class.fixture_path, "test-fixture.script")
     # Level names match those in 'test.script'
     @levels = (1..5).map {|n| create(:level, name: "Level #{n}", game: @game)}
 
-    Rails.application.config.stubs(:levelbuilder_mode).returns false
+    @course = create(:course)
+    @script_in_course = create(:script, hidden: true)
+    create(:course_script, position: 1, course: @course, script: @script_in_course)
+
+    # ensure that we have freshly generated caches with this course/script
+    Course.clear_cache
+    Script.clear_cache
   end
 
   def populate_cache_and_disconnect_db
     Script.stubs(:should_cache?).returns true
     # Only need to populate cache once per test-suite run
     @@script_cached ||= Script.script_cache_to_cache
-    Script.script_cache_from_cache
+    Script.script_cache
```

Comment:
Script.script_cache will call Script.script_cache_from_cache, but by calling Script.script_cache instead we ensure that @@script_cache in Script has been populated.